### PR TITLE
Deploy authentication component to stone-prod-p01

### PR DESCRIPTION
**Pull request by hugares**
Authentication component now supports Red Hat IdP so deploy it to stone-prod-p01 to get rover groups synced and  grant permissions to different groups based on existing RBACs.

[RHTAPSRE-287](https://issues.redhat.com//browse/RHTAPSRE-287)

### CODE REVIEW

File: components/authentication/production/base/kustomization.yaml [hunk 1]
```diff
@@ -2,6 +2,8 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 resources:
   - ../../base/github
+components:
+  - ../k-components/rhtap-infra-secrets-patch
```
Comment (gbenhaim) [inline components/authentication/production/base/kustomization.yaml:6]: what is the benefit of putting it in a component?

Comment (hugares) [reply components/authentication/production/base/kustomization.yaml:6]: that patch is used in both `production/base` and `production/stone-prod-p01`, I did not want to duplicate the patch

Comment (hugares) [reply components/authentication/production/base/kustomization.yaml:6]: hold on, I copied this from somewhere to share the patch but IIUC your comment, I could do the same with a resources. Let me test something

Comment (hugares) [reply components/authentication/production/base/kustomization.yaml:6]: If I convert the rhtap-infra-secrets-patch from a component to a patch, it does not work. The patch is applied before the resource to be patched is loaded.

**Pull request merged**